config/src/config/node_config_loader.rs

```diff
@@ -14,8 +15,48 @@ use aptos_types::{
     transaction::{Transaction, WriteSetPayload},
     write_set::WriteOp,
 };
+use serde_yaml::Value;
 use std::path::Path;
 
+/// A simple enum to represent the type of a node
+/// as determined from the config file.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum NodeType {
```

Comment:
We will have one too many `NodeType`s. Should we converge in some way?

https://github.com/aptos-labs/aptos-core/blob/5af60240f125b2596ebd93ceb52fad29f284bbd9/crates/aptos-telemetry-service/src/types/mod.rs#L44

Reply:
See @gregnazario's comment above. The problem with merging them is that to do it properly, it'll be a breaking change, and we'll need to update the docs and our various config template. I'm not against doing it, but we'll need to do it separately. Note that the new type is internal only to the sanitizer/optimizer for now. Does that make sense?